odoo/tools/pdf.py

```diff
@@ -78,6 +83,65 @@ def rotate_pdf(pdf):
         writer.write(_buffer)
         return _buffer.getvalue()
 
+
+def add_banner(pdf_stream, text=None, thickness=2 * cm):
+    """ Add a banner on a PDF, in the upper right corner with Odoo's logo.
+
+    :param pdf_stream (BytesIO):    The PDF stream where the banner will be applied.
+    :param text (str):              The text to be displayed.
+    :param thickness (float):       The thickness of the banner in pixels.
+    :return (BytesIO):              The modified PDF stream.
+    """
+
+    old_pdf = PdfFileReader(pdf_stream, strict=False, overwriteWarnings=False)
+    old_pdf.getNumPages()
+    packet = io.BytesIO()
+    can = canvas.Canvas(packet)
+    odoo_logo = Image.open(file_open('base/static/img/main_partner-image.png', mode='rb'))
```

Comment:
Add the stream to/file descriptor of the image as an argument, defaulting to None to give the ability of adding a banner without a logo